railties/lib/rails/generators/rails/scaffold_controller/templates/controller.rb

```diff
@@ -49,7 +77,27 @@ def destroy
     @<%= orm_instance.destroy %>
     redirect_to <%= index_helper %>_url, notice: <%= "'#{human_name} was successfully destroyed.'" %>
   end
+  <% if collection_routing? %>
+  # Collection routes
+  # PATCH <%= route_url %>/1..10
+  def update_many
+    <%= orm_class.find(class_name, "#{singular_table_name}_ids") %>.each do |<%= singular_table_name %>|
+      <%= singular_table_name %>.update(params[:<%= plural_table_name %>][:"#{<%= singular_table_name %>.id}"])
+    end
+    redirect_to @<%= singular_table_name %>, notice: <%= "'#{human_name}s were successfully updated.'" %>
+  end
 
+  # PUT <%= route_url %>/1..10
+  def replace
+    <%= orm_class.destroy_all(class_name) %>
+    create
+  end
```

Comment:
Let's review the benefits of this PR in the context of real-world Rails apps and find one good use-case for each one of them.

##### GET /posts/:ids

I display a list of items in a table and I paginate them, so I only display 10 items per page.

##### DELETE /posts/:ids

I display a list of items in a table. I have a check-box to select each one of them and I have a button under the table saying "Delete selected items".

##### PATCH  /posts/:ids 

I display a list of items in a table. I can double click on each item to turn their description into a text field (via Javascript) and I have a button under the table saying "Save changes".

##### POST /posts {:collection=>true}

I display an inline form to add items. A "+" button lets me add (via Javascript) more fields so I can enter multiple items and submit them at once.

##### PUT  /posts/:ids 

I… cannot think of a good example. @ujjwalt can you come up with one?